giskard/core/suite.py

```diff
@@ -435,10 +441,15 @@ def upload(self, client: GiskardClient, project_key: str):
             if isinstance(arg, BaseModel) or isinstance(arg, Dataset):
                 _try_upload_artifact(arg, client, project_key, uploaded_uuid_status)
 
-        self.id = client.save_test_suite(self.to_dto(client, project_key, uploaded_uuid_status))
+        if self.id:
+            client.update_test_suite(self.id, self.to_dto(client, project_key, uploaded_uuid_status))
+            analytics.track("hub:test_suite:updated")
+        else:
+            self.id = client.save_test_suite(self.to_dto(client, project_key, uploaded_uuid_status))
+            analytics.track("hub:test_suite:uploaded")
 
         print(f"Test suite has been saved: {client.host_url}/main/projects/{project_key}/test-suite/{self.id}/overview")
-        analytics.track("hub:test_suite:uploaded")
```

Comment:
Should we track in both operations: creation by uploading and update by uploading?

Reply:
Yes it has been moved to line 449